todos-app/src/App/TodoList/TodoList.tsx

```diff
@@ -1,40 +1,46 @@
-import React, { ReactElement, useState } from "react";
-import { ITodo } from "../../DataStructure";
+import { useEffect, useState } from "react";
+import { Todo } from "../models/Todos";
 import TodoListItem from "./TodoListItem/TodoListItem";
 
-type TodoListProps = {
-  todos: ITodo[];
+type Props = {
+  todos: Todo[];
   onToggle(id: string): void;
   onRemove(id: string): void;
 };
 
-const TodoList: React.FC<TodoListProps> = ({ todos, onToggle, onRemove }) => {
-  const toggleAllCheckbox = (): void => {
+const TodoList = ({ todos, onToggle, onRemove }: Props) => {
+  const [isChecked, setIsChecked] = useState(false);
+  let toggleAllState = todos.every((todo) => {
+    return todo.completed === true;
+  });
+
+  useEffect(() => {
     todos.forEach((todo) => {
-      onToggle(todo.id);
+      if (todo.completed === toggleAllState) {
```

Comment:
I'm a bit confused, when I click on "mark all completed" why does it need to check if all todo's are completed, would you simply filter todos that are not completed and mark them as complete? something like: 
```
todos
  .filter(!todo.completed)
  .forEach(todo => onToggle(todo.id))
```

Reply:
If I don't write .filter((todo) => !todo.completed) and etc it does not work correctly, it could not untoggle all todos. So I left this like that:
     _todos
      .filter((todo) => todo.completed === allCompleted)
      .forEach((todo) => onToggle(todo.id));_

Reply:
I think re-writing that without using useEffect as mentioned in the comment below will simplify it